[5.8] PHPUnit 8 support

driesvints opened the pull request:
**This PR needs to wait until orchestra/testbench-core has PHPUnit 8 support: https://github.com/orchestral/testbench-core/pull/19**

This adds support for PHPUnit 8 to the next release while keeping backwards compatibility with PHPUnit 7. The only necessary changes made which will affect user land are the now required return types on the `setUp` and `tearDown` methods. This will affect both PHPUnit 7 and PHPUnit 8 users and cannot be avoided.

This PR also replaces all deprecated methods which will be removed in PHPUnit 9. The only call that remains is the `assertArraySubset` call in the `assertJson` method (which other methods depend on). We'll have to either take over te method from PHPUnit or deprecate and remove these.

driesvints committed:
Allow install of PHPUnit 8

driesvints committed:
Add void return type to setUp and tearDown methods

driesvints committed:
Replace expect exception annotations

driesvints committed:
Replace incorrect string assertions with assertContains

driesvints committed:
Apply fixes from StyleCI (#27440)

crynobone commented:
Please update composer.json to

```json
    "orchestra/testbench-core": "dev-phpunit8 as 3.8.x-dev"
```

driesvints committed:
Use dev-phpunit8 for testing

driesvints committed:
Fix MailLogTransportTest

driesvints committed:
Fix MailSesTransportTest

driesvints committed:
Take over assertArraySubset from PHPUnit

This method which is deprecated now throws warnings on usage causing builds to fail. We'll have to take over the method ourselves if we want to keep using it.

driesvints commented:
@crynobone Done! Build passes now. Should we merge first?

mnabialek commented on tests/Auth/AuthDatabaseTokenRepositoryTest.php:
```diff
@@ -13,14 +13,14 @@
 class AuthDatabaseTokenRepositoryTest extends TestCase
 {
-    public function setUp()
+    public function setUp(): void
```
As we are doing such big refactoring, probably it's a good moment to make all `setUp` and `tearDown` methods protected as they are in PHPUnit. Now some of them are public and others are protected and there is no point to make them public.

driesvints replied:
I realized I forgot to comment here but I've updated everything as such. Thanks for mentioning 👍
